locust/main.py

```diff
@@ -344,11 +345,24 @@ def timelimit_stop():
         spawn_run_time_limit_greenlet()
 
     stats_printer_greenlet = None
+    input_listener_greenlet = None
     if not options.only_summary and (options.print_stats or (options.headless and not options.worker)):
         # spawn stats printing greenlet
         stats_printer_greenlet = gevent.spawn(stats_printer(runner.stats))
         stats_printer_greenlet.link_exception(greenlet_exception_handler)
 
+        input_listener_greenlet = gevent.spawn(
```

Comment:
Is this really the right place to register it? Why not register it all the time (before this if statement, maybe skipping options.worker)

The current way, running locust with --only-summary, would silently disable key commands (which is weird)

And I dont think there is any down side to having it running even in web mode (although you might need to verify that locust is actually running before someone tries to ramp up/down)